Composer/packages/lib/bot-deploy/src/botProjectDeploy.ts

```diff
@@ -0,0 +1,597 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { WebSiteManagementClient } from '@azure/arm-appservice-profile-2019-03-01-hybrid';
+import { ResourceManagementClient } from '@azure/arm-resources';
+import {
+  Deployment,
+  DeploymentsCreateOrUpdateResponse,
+  DeploymentsValidateResponse,
+  ResourceGroup,
+  ResourceGroupsCreateOrUpdateResponse,
+} from '@azure/arm-resources/esm/models';
+import { GraphRbacManagementClient } from '@azure/graph';
+import * as msRestNodeAuth from '@azure/ms-rest-nodeauth';
+import * as fs from 'fs-extra';
```

Comment:
I'm concerned about this library being bound to being used on the filesystem. We should hook into the storage layer of Composer to reuse the serialize configurations when writing .dialog, .lu, .lg, so when someone has CosmosDB as its storage layer, this library can still service their Azure deployment needs. 

@benbrown @carlosscastro Looking for your suggestions on priority. Do we P0 or P1 this? The work to pull this out is non-trivial and could take awhile. Perhaps we land Publishing & Runtime E2E and revisit.

Reply:
i need to get deeper into how all of this works, but as it is, we pull stuff *out of storage and onto disk* before doing the deploy -- so in that sense, it doesn't really matter.